### PR TITLE
ipn/ipnlocal, util/winutil/policy: modify Windows profile migration t…

### PR DESCRIPTION
**Pull request by dblohm7**
…o load legacy prefs from within tailscaled

I realized that a lot of the problems that we're seeing around migration and LocalBackend state can be avoided if we drive Windows pref migration entirely from within tailscaled. By doing it this way, tailscaled can automatically perform the migration as soon as the connection with the client frontend is established.

Since tailscaled is already running as LocalSystem, it already has access to the user's local AppData directory. The profile manager already knows which user is connected, so we simply need to resolve the user's prefs file and read it from there.

Of course, to properly migrate this information we need to also check system policies. I moved a bunch of policy resolution code out of the GUI and into a new package in util/winutil/policy.

Updates #7626

### CODE REVIEW

File: ipn/ipnlocal/profiles.go [hunk 1]
```diff
@@ -66,7 +65,13 @@ func (pm *profileManager) SetCurrentUserID(uid ipn.WindowsUserID) error {
 	// the selected profile for the current user.
 	b, err := pm.store.ReadState(ipn.CurrentProfileKey(string(uid)))
 	if err == ipn.ErrStateNotExist || len(b) == 0 {
-		pm.NewProfile()
+		if runtime.GOOS == "windows" {
+			if err := pm.migrateFromLegacyPrefs(); err != nil {
```
Comment (maisem) [inline ipn/ipnlocal/profiles.go:69]: I think you need to handle `errAlreadyMigrated`?

File: ipn/ipnlocal/profiles_windows.go [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+		return "", err
+	}
+	if usr.HomeDir == "" {
+		return "", fmt.Errorf("user %q does not have a home directory", uid)
+	}
+	userLegacyPrefsDir := filepath.Join(usr.HomeDir, "AppData", "Local", "Tailscale")
+	return userLegacyPrefsDir, nil
+}
+
+type ctxKeyMigrationSentinelPath struct{}
```
Comment (maisem) [inline ipn/ipnlocal/profiles_windows.go:42]: I don't understand why we need a context here. just return the path the sentinel file instead?

Comment (dblohm7) [reply ipn/ipnlocal/profiles_windows.go:42]: Sure.

File: ipn/ipnlocal/profiles.go (hunk 1 above)
Comment (dblohm7) [reply ipn/ipnlocal/profiles.go:69]: @maisem I guess the question is, what can we actually do to handle it at this point?

By reaching this point, we've already established that the current profile key is not yielding any profile info, and yet the migration code is telling us that we've already migrated. Given the inconsistency of that state, I'm not sure what else we can do other than propagate that message up the stack. Thoughts?

File: ipn/ipnlocal/profiles_windows.go [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+		return "", err
+	}
+	if usr.HomeDir == "" {
+		return "", fmt.Errorf("user %q does not have a home directory", uid)
+	}
+	userLegacyPrefsDir := filepath.Join(usr.HomeDir, "AppData", "Local", "Tailscale")
+	return userLegacyPrefsDir, nil
+}
+
+func (pm *profileManager) loadLegacyPrefs() (string, ipn.PrefsView, error) {
```
Comment (maisem) [inline ipn/ipnlocal/profiles_windows.go:41]: document that the returned string is the `migrationSentinel` file name

File: ipn/ipnlocal/profiles.go (hunk 1 above)
Comment (maisem) [reply ipn/ipnlocal/profiles.go:69]: good point, i guess it should error out

**Review by maisem — APPROVED** [review]

**Pull request merged**